packages/expect/src/jest-expect.ts

```diff
@@ -1045,7 +1045,7 @@ export const JestChaiExpect: ChaiPlugin = (chai, utils) => {
             return result instanceof chai.Assertion ? proxy : result
           }
 
-          return async (...args: any[]) => {
+          return (...args: any[]) => {
```

Comment:
I think I understand now. When returning promise from `async () => { return somePromise }` and `somePromise` rejects, even if we handle rejections (e.g. by `somePromise.catch` or `Promise.allSettlled([somePromise])`, `somePromise` rejection still leaks to the promise returned from `async` function calls.